quic/s2n-quic/src/tests/dc.rs

```diff
@@ -110,35 +124,46 @@ fn self_test<S: ServerProviders, C: ClientProviders>(
 
         let addr = server.local_addr()?;
         spawn(async move {
-            let mut conn = server.accept().await.unwrap();
-            let mut stream = conn.open_bidirectional_stream().await.unwrap();
-            stream.send(vec![42; LEN].into()).await.unwrap();
-            stream.flush().await.unwrap();
+            let conn = server.accept().await;
+            if expected_error.is_some() {
+                assert!(conn.is_none());
+            } else {
+                // Allow the client to close the connection first
+                let _ = conn.unwrap().accept().await;
+            }
         });
 
         let client = client
             .with_io(handle.builder().build().unwrap())?
-            .with_event((tracing_events(), client_subscriber))?
+            .with_event((ClientConfirm, (tracing_events(), client_subscriber)))?
             .with_random(Random::with_seed(456))?
             .with_dc(MockDcEndpoint::new(&client_tokens))?
             .start()?;
 
         primary::spawn(async move {
             let connect = Connect::new(addr).with_server_name("localhost");
             let mut conn = client.connect(connect).await.unwrap();
-            let mut stream = conn.accept_bidirectional_stream().await.unwrap().unwrap();
-
-            let mut recv_len = 0;
-            while let Some(chunk) = stream.receive().await.unwrap() {
-                recv_len += chunk.len();
+            let result = ClientConfirm::wait_ready(&mut conn).await;
+
+            if let Some(error) = expected_error {
+                assert_eq!(error, result.err().unwrap().kind());
+            } else {
+                assert!(result.is_ok());
+                // wait briefly so the ack for the `DC_STATELESS_RESET_TOKENS` frame from the server is sent
+                // before the client closes the connection. This is only necessary to confirm the `dc::State`
+                // on the server moves to `DcState::Complete`
+                delay(Duration::from_millis(100)).await;
```

Comment:
Can we make an assertion that the client has completed the `dc` state here before we delay? In real pathways we obviously won't be sleeping.